app/frontend/src/features/profile/edit/EditHostingPreference.test.tsx

```diff
@@ -0,0 +1,47 @@
+import {
+  render,
+  screen,
+  waitFor,
+  queryByAttribute,
+  fireEvent,
+  act,
+} from "@testing-library/react";
+import { Route, Switch } from "react-router-dom";
+import { addDefaultUser, MockedService } from "../../../test/utils";
+import wrapper from "test/hookWrapper";
+import EditHostingPreference from "./EditHostingPreference";
+import { getUser } from "test/serviceMockDefaults";
+import { service } from "service";
+import { profileRoute } from "routes";
+
+const MockProfilePage = "Mock Profile Page";
+const getUserMock = service.user.getUser as MockedService<
+  typeof service.user.getUser
+>;
+const MockProfileComponent = () => <div>{MockProfilePage}</div>;
+
+describe("EditHostingPreference", () => {
+  beforeEach(() => {
+    getUserMock.mockImplementation(getUser);
+    addDefaultUser();
+  });
+
+  it("should redirect to Profile route after successful update", async () => {
+    const { getByText } = render(
+      <Switch>
+        <Route path={profileRoute}>
+          <MockProfileComponent />
+        </Route>
+        <Route>
+          <EditHostingPreference />
+        </Route>
+      </Switch>,
+      { wrapper }
+    );
+    const saveButton = await waitFor(() => getByText(/save/i));
+    const rightClick = { button: 2 };
+    fireEvent.click(saveButton, rightClick);
```

Comment:
```suggestion
    userEvent.click(await screen.findByRole('button', { name: SAVE }));
```
A couple of things here:
- In React Testing Library (RTL), `*ByRole` should be preferred over `*ByText` since it also gives us a basic idea on how accessible our components are - [more info here](https://testing-library.com/docs/queries/about/#priority)
- `findBy*` is a shorthand for the `waitFor(() => ...)` and `getBy*` combo you wrote there
-  right click is unnecessary? The interaction here is triggered by a left click and RTL does that by default
- Prefer using `userEvent` from `@testing-library/user-event` [for reasons explained here](https://testing-library.com/docs/queries/about/#priority)
- Since we are not using the button for other purpose than clicking, I think it's clear enough to combine the two steps together

Reply:
Also `SAVE` is available as a constant in `features/constants`

Reply:
Thanks for the comments with appropriate links as well. You are awesome. 👍🏻